akka-http-core/src/main/scala/akka/http/impl/engine/parsing/BodyPartParser.scala

```diff
@@ -290,12 +291,11 @@ private[http] object BodyPartParser {
   }
 
   // TODO: load from config
-  val defaultSettings = Settings(
-    maxHeaderNameLength = 64,
-    maxHeaderValueLength = 8192,
-    maxHeaderCount = 64,
-    illegalHeaderWarnings = true,
-    headerValueCacheLimit = 8,
-    uriParsingMode = Uri.ParsingMode.Relaxed,
-    cookieParsingMode = ParserSettings.CookieParsingMode.RFC6265)
+
+  // TODO: load from config
+  object Settings {
+    def apply(s: ParserSettings): Settings = Settings(s.maxHeaderNameLength, s.maxHeaderValueLength, s.maxHeaderCount, s.illegalHeaderWarnings, s.headerValueCacheLimit("default"), s.uriParsingMode, s.cookieParsingMode)
+  }
+
+  val defaultSettings = Settings(ParserSettings.fromSubConfig(ConfigFactory.load, ConfigFactory.systemProperties()))
```

Comment:
ohh yes. It did seem strange while I did it. 

However BodyPartParser constructor can take a Settings object and is set get the defaultSettings if nothing is passed

```
private[http] final class BodyPartParser(defaultContentType: ContentType,
                                         boundary: String,
                                         log: LoggingAdapter,
                                         settings: BodyPartParser.Settings = BodyPartParser.defaultSettings)
```

I can replace the inner object to take 
`actorSystem.settings.config` instead of the `ConfigFactory.systemProperties()`

Is this the right direction?